erpnext/buying/doctype/supplier/supplier.py

```diff
@@ -153,3 +160,22 @@ def get_supplier_primary_contact(doctype, txt, searchfield, start, page_len, fil
 		""",
 		{"supplier": supplier, "txt": "%%%s%%" % txt},
 	)
+
+
+def get_users_query():
+	user = frappe.qb.DocType("User")
+	hasRole = frappe.qb.DocType("Has Role")
+	q = (
+		frappe.qb.from_(user)
+		.inner_join(hasRole)
+		.on(user.name == hasRole.parent)
+		.select(user.name, user.full_name)
+		.where(hasRole.role == "Supplier")
```

Comment:
If possible lets use `frappe.get_list` instead of raw query. Raw query wont respect permission rules that site might have configured. 


Something like this will work:
`frappe.get_list("User", {"role": "Supplier"}, ["name", "full_name"])`

Also, no need for two separate functions here IMO. Original function can call get_list directly.